sources/web/datalab/static.ts

```diff
@@ -112,6 +116,23 @@ function datalabFileExists(filePath: string) {
 }
 
 /**
+ * Sends a static 'custom.css' file located within the user's config directory.
+ *
+ * @param userId the ID of the current user.
+ * @param response the out-going response associated with the current HTTP request.
+ */
+function sendUserCustomTheme(userId: string, response: http.ServerResponse): void {
+    var customThemePath = path.join(settings.getUserConfigDir(userId), CUSTOM_THEME_FILE);
+    fs.stat(customThemePath, function(e, stats) {
```

Comment:
Actually, I'm now wondering if we should have an entirely separate cache for this type of dynamic file.

Right now, the only way to force a cache flush is to restart Datalab, and if a user is trying to tweak their custom theme, then that results in a very slow and cumbersome iterative development process.

What we could do instead is store both the file contents and it's metadata in the cache, then always run 'fs.stat' before looking in the cache. If we see the same timestamp then return the cached contents otherwise re-read the file, cache the updated contents, and then return them.

That would mean we always run fs.stat for reading these dynamic files, but would give the user an easier feedback loop where after saving their changes they just have to reload the webpage.

What do you think?